sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADOAuth2OboAuthorizedClientRepository.java

```diff
@@ -87,6 +96,29 @@ public AADOAuth2OboAuthorizedClientRepository(ClientRegistrationRepository repos
             request.setAttribute(oboAuthorizedClientAttributeName, (T) oAuth2AuthorizedClient);
             return (T) oAuth2AuthorizedClient;
         } catch (Throwable throwable) {
+            // Handle conditional access policy for obo flow.
+            String claims = Optional.of(throwable)
+                                    .map(Throwable::getCause)
+                                    .filter(e -> e instanceof MsalInteractionRequiredException)
+                                    .map(e -> (MsalInteractionRequiredException) e)
+                                    .map(MsalInteractionRequiredException::claims)
+                                    .orElse(null);
+
+            if (claims != null) {
+                ServletRequestAttributes attr =
+                    (ServletRequestAttributes) RequestContextHolder.currentRequestAttributes();
+                HttpServletResponse response = attr.getResponse();
+                assert response != null;
```

Comment:
Use `Assert.notNull()` instead of `assert`. `assert` should not be used in production code.

Refs: https://stackoverflow.com/questions/2758224/what-does-the-java-assert-keyword-do-and-when-should-it-be-used